File: leetcode3/염혜정/3754. Concatenate Non-Zero Digits and Multiply by Sum I.java
// x = 0을 제외한 값
// sum = x의 각 값을 더한 값

class Solution {
    public long sumAndMultiply(int n) {
        if (n == 0) return 0;
        
        String s = "";
        long sum = 0;
        while (n>0) {
            int digit = n % 10;
            if (digit != 0) {
                s = digit + s;
                sum += digit;
            }
            n /= 10;
        }
        long x = Long.parseLong(s);
        return x * sum;
    }
}

File: leetcode3/남효정/3754. Concatenate Non-Zero Digits and Multiply by Sum I.py

'''
1. 아이디어 :
2. 시간복잡도 :
    O(n)
3. 자료구조/알고리즘 :
'''

class Solution:
    def sumAndMultiply(self, n: int) -> int:
        if n == 0: return n
        
        non_zero = str(n).replace('0', '')
        sum_val = sum([int(i) for i in non_zero])
        x = int(non_zero)

        return x * sum_val